src/main/java/biz/paluch/logging/gelf/jul/JulLogEvent.java

```diff
@@ -104,13 +104,10 @@ private String getThreadName(LogRecord record) {
 
         long threadId = record.getThreadID();
         String threadName = "" + record.getThreadID();
-        ThreadMXBean threadMXBean = ManagementFactory.getThreadMXBean();
-        for (long id : threadMXBean.getAllThreadIds()) {
-            if (id == threadId) {
-                ThreadInfo threadInfo = threadMXBean.getThreadInfo(id);
-                if (threadInfo != null) {
-                    threadName = threadInfo.getThreadName();
-                }
+        Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
```

Comment:
`getAllStackTraces()` is quite expensive. Given that we have a cache in place, we should soften the actual impact.

We should be able to optimize for the default logger arrangement. Typically, log handlers are called on the same thread as the logging happens. Therefore, we should call `Thread.currentThread().getId()` first to inspect whether we need to obtain all threads or whether we can resolve the thread without further do.

Reply:
done

Reply:
done